pkg/azure/provider.go

```diff
@@ -249,6 +254,27 @@ func (p *Provider) GetServicePrincipalToken(env *azure.Environment, resource str
 		err = fmt.Errorf("nmi response failed with status code: %d", resp.StatusCode)
 		return nil, err
 	}
+
+	if p.UseVMManagedIdentity {
+		msiEndpoint, err := adal.GetMSIVMEndpoint()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get managed identity (MSI) endpoint")
+		}
+
+		if p.UserAssignedIdentityID != "" {
+			log.Infof("azure: using user assigned managed identity %s to retrieve access token", p.UserAssignedIdentityID)
```

Comment:
we should redact the clientID while logging similar to - https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/pkg/azure/provider.go#L233

I think we can add a function to redact clientID and reuse it here